arex-agent-bootstrap/src/main/java/io/arex/agent/bootstrap/util/AdviceClassesCollector.java

```diff
@@ -95,6 +102,70 @@ private void addClassToInjectorCache(String adviceClassName) {
+            addURL.setAccessible(true);
+
+            if (classLoader instanceof URLClassLoader) {
+                addURL.invoke(classLoader, jarFile.toURI().toURL());
+            }
+
+            /*
+             * Due to Java 8 vs java 9+ incompatibility issues
+             * See https://stackoverflow.com/questions/46694600/java-9-compatability-issue-with-classloader-getsystemclassloader/51584718
+             */
+            ClassLoader urlClassLoader = ClassLoader.getSystemClassLoader();
+            if (!(urlClassLoader instanceof URLClassLoader)) {
+                try (URLClassLoader tempClassLoader = new URLClassLoader(new URL[] {jarFile.toURI().toURL()}, urlClassLoader)) {
+                    addURL.invoke(tempClassLoader, jarFile.toURI().toURL());
+                    appendToAppClassLoaderSearch(classLoader, jarFile);
+                }
+            } else {
+                addURL.invoke(urlClassLoader, jarFile.toURI().toURL());
+                appendToAppClassLoaderSearch(classLoader, jarFile);
+            }
+        } catch (Exception e) {
+            System.err.printf("appendToClassLoaderSearch failed, classLoader: %s, jarFile: %s%n",
+                    classLoader.getClass().getName(), jarFile.getAbsolutePath());
+        }
+    }
+
+    /**
+     * append jar jdk.internal.loader.ClassLoaders.AppClassLoader
+     * if java >= 11 need add jvm option:--add-opens=java.base/jdk.internal.loader=ALL-UNNAMED
+     * @param classLoader
+     * @param jarFile
+     */
+    private static void appendToAppClassLoaderSearch(ClassLoader classLoader, File jarFile) {
+        try {
+            Class<? extends ClassLoader> loaderClass = classLoader.getClass();
+            if (JdkUtils.isJdk11() && ConfigConstants.APP_CLASSLOADER_NAME.equalsIgnoreCase(loaderClass.getName())) {
+                Method classPathMethod = loaderClass.getDeclaredMethod("appendToClassPathForInstrumentation", String.class);
+                classPathMethod.setAccessible(true);
+                classPathMethod.invoke(classLoader, jarFile.getPath());
+            }
+        } catch (Exception e) {
+            System.err.printf("appendToAppClassLoaderSearch failed, classLoader: %s, jarFile: %s%n",
+                    classLoader.getClass().getName(), jarFile.getAbsolutePath());
+        }
+    }
+
+    public void addSpecificJarToLoaderSearch(String jarName) {
+        String thirdPartyJarFilePaths = System.getProperty(ConfigConstants.THIRD_PARTY_JAR_FILE_PATH);
+        if (StringUtil.isEmpty(thirdPartyJarFilePaths)) {
+            return;
+        }
+        String[] split = StringUtil.split(thirdPartyJarFilePaths, ';');
+        for (String thirdPartyJarFilePath : split) {
+            if (StringUtil.containsIgnoreCase(thirdPartyJarFilePath, jarName)) {
+                appendToClassLoaderSearch(Thread.currentThread().getContextClassLoader(), new File(thirdPartyJarFilePath));
+            }
```

Comment:
The assembly dependencySet output directory can be `/third-parth/jackson`.

We can use the folder name as the mapping key, put `arex.third.parth.xxx.path` into the system properties will get it directly without filtering it, such as `arex.third.parth.jackson.path`